docs/react/week-3/learning-objectives.md

```diff
@@ -6,8 +6,14 @@ sidebar_label: Learning Objectives
 
 ## React Week 3
 
-### Data fetching (advanced)
+### Data fetching
 
+Show some data from an API in a component.
+
+- Can explain why a loading state is necessary when fetching data as the result is not available on the initial render
+- Be able to trigger `fetch()` using the `useEffect()` callback, with an empty array (`[]`) as the 2nd dependencies argument
+- Be able to store data from a successful API call in state
+- Be able to catch an unsuccessful API call, store the error in state and render an error message
 - Allow effects to update in response to prop changes
   - Explain why a component with an effect dependent on props is broken with empty `useEffect()` dependencies (`[]`)
   - Be able to fix a component with an effect dependent on props using the `useEffect()` dependencies
```

Comment:
Below: This class didn't cover `onSubmit` at all, and in particular didn't cover `event.preventDefault()` - should it?

Reply:
Yep it probably should, but again concerned about the time.

Reply:
This _is_ one I've seen people struggle with in exercises, particularly when they're working on the Chat App later on... Do you think we could squeeze it into week 2 as a 10 minute exercise which follows on from click handlers? Maybe a "Give my caught pokemon a nickname" form? We can hand-wave over `event.preventDefault()` by just saying something like "We need to do this whenever we handle a form in React, if you're interested in why, [here](https://www.robinwieruch.de/react-preventdefault)'s some further reading"